.pre-commit-config.yaml

```diff
@@ -0,0 +1,60 @@
+repos:
```

Comment:
I think it is kind of unfortunate to duplicate versions and configurations here --- all we really need is like a 4-line shell script that invokes pylint, mypy, black, npm run check.  I guess pre-commit creates its own venv used only for running the checks?

Reply:
>  I guess pre-commit creates its own venv used only for running the checks?

yep.

The added benefits I find with pre-commit is that I don't have to on the look out for
- CRLFs
- trailing whitespace
- yaml/toml/json syntax errors (which will always seem to crop up)
- EoF is a single blank line

I would not prefer to have an in-house script because it is just more work to maintain. I personally can't use a .sh script as that isn't friendly with Windows (without an env created in WSL which is significantly slower). Using pre-commit also provides some useful feedback in the event that a check failed.

Having the pre-commit config filled out also shows what deps are needed for which check. I suppose we could keep comments in the dev-requirements.txt file about this, but that is more of a nuisance to me.

---
In re-writing the CI, I noticed that the only thing needed from dev-requirements.txt is `wheel`, so that file is no longer used in CI.